crates/dprint/src/paths.rs

```diff
@@ -91,12 +91,24 @@ pub async fn get_and_resolve_file_paths<'a>(
 ) -> Result<GlobOutput> {
   let cwd = environment.cwd();
   let mut file_patterns = get_all_file_patterns(config, args, &cwd);
+
+  if args.only_staged {
+    if let Ok(staged_files) = environment.get_staged_files() {
```

Comment:
I think we should surface the error? Otherwise people might be confused about the behaviour when this fails.